src/alchemlyb/parsing/util.py

```diff
@@ -9,15 +9,23 @@
 # need to do some backflips to support Python 2 bz2 behavior
 # bz2 in Python 2 doesn't have an open function, and in Python 3
 # the BZ2File class only does binary mode
+# similar changes need to be made for gzip
+# gzip in Python 2 doesn't have a text mode, and in Python 3
+# gzip does have text mode
 try:
     bz2.open
 except AttributeError:
     bz2_open = bz2.BZ2File
+    gzip_open = gzip.open
```

Comment:
Please put this in a separate try/except statement that does not depend on bz2. Decouple everything. Find a different way to test for gz specifically. `six` also has ways to tell you if it is Py 2 or Py 3 – use those if nothing else works.